nexus/src/app/instance.rs

```diff
@@ -62,56 +62,28 @@ impl super::Nexus {
         instance_selector: &'a params::InstanceSelector,
     ) -> LookupResult<lookup::Instance<'a>> {
         match instance_selector {
```

Comment:
You'll notice this selector logic is greatly reduced. It checks the name and ID case of the resource being selected against and delegates selection of sub resources to that resource's lookup.

Reply:
That looks a lot cleaner. The error cases will likely need to be expanded depending on how that gets spec'd. Specifically the over-specification case will want a different error message.

For example:
```rust
InstanceSelector { project_selector: Some(project_selector), instance: NameOrId::Id(id) }
```

Reply:
Right, yeah. @david-crespo  mentioned in the RFD how he thought that case shouldn't be an error and I'm still undecided. Should we just ignore that case?

Reply:
Fair point. I think disallowing invalid combinations is the lowest common denominator.

Reply:
I've updated this in https://github.com/oxidecomputer/omicron/pull/2050/commits/69ac00809a03f69a323a74986f9e8deb3af12abe to have expanded error checks. Essentially I just added an extra case for instance and project.